crates/nu-parser/src/path.rs

```diff
@@ -70,21 +95,81 @@ pub fn expand_path<'a>(path: &'a str) -> Cow<'a, str> {
 mod tests {
     use super::*;
 
+    // common tests
     #[test]
     fn string_without_ndots() {
         assert_eq!("../hola", &expand_ndots("../hola").to_string());
     }
 
+    #[test]
+    fn string_with_three_ndots_and_chars() {
+        assert_eq!("a...b", &expand_ndots("a...b").to_string());
+    }
+
+    #[test]
+    fn string_with_two_ndots_and_chars() {
+        assert_eq!("a..b", &expand_ndots("a..b").to_string());
+    }
+
+    #[test]
+    fn string_with_one_dot_and_chars() {
+        assert_eq!("a.b", &expand_ndots("a.b").to_string());
+    }
+
+    // Windows tests
+    #[cfg(windows)]
     #[test]
     fn string_with_three_ndots() {
-        assert_eq!("../..", &expand_ndots("...").to_string());
+        assert_eq!(r"..\..", &expand_ndots("...").to_string());
+    }
+
+    #[cfg(windows)]
+    #[test]
+    fn string_with_mixed_ndots_and_chars() {
+        assert_eq!(
+            r"a...b/./c..d/../e.f/..\..\..//.",
+            &expand_ndots("a...b/./c..d/../e.f/....//.").to_string()
+        );
+    }
+
+    #[cfg(windows)]
+    #[test]
+    fn string_with_three_ndots_and_final_slash() {
+        assert_eq!(r"..\../", &expand_ndots(".../").to_string());
```

Comment:
For the final slashes on Windows, should these go in the direction that's native to the platform?

Reply:
Currently, what the `expand_ndots()` does is replace manydots (`....`) with a series of `..` (`../..`). Everything else is noop. So the final slash is out of scope of the code, it simply copy-pastes whatever the input is.

Reply:
At this point, the path is just a &str, so it doesn't know about filenames yet. The `expand_ndots()` just replaces any `...` with `../..` and that's all it does. If the user put `...\` as an input, the result would be `..\..\`.

Reply:
I'm not exactly sure what you're asking here.